controllers/AcquisitionSessionController.php

```diff
@@ -0,0 +1,291 @@
+        $existingFilePath = $this->getTemplateFile();
+        if ($this->error != null) {
+            return $this->error;
+        }
+        // add data retreive from the ws to the worksheet choosen
+        $newFilePath = $this->addHiddenphisSheetData($existingFilePath);
+        if ($this->error != null) {
+            return $this->error;
+        }
+        // save the new file and send it to the user
+        $this->sendTemplateFile($newFilePath);
+        if ($this->error != null) {
+            return $this->error;
+        }
+    }
+
+    /**
+     * Retreive the latest saved document which matches
+     * with the choosen document type
+     * @return mixed null
+     *               or a string with the saved document path
+     */
+    private function getTemplateFile() {
+        // 1. Get installation uri
+        $infrastructureUri = substr(Yii::$app->params['baseURI'], 0, -1);
+        
+        // 2. Get the last acquisition template for the document type required
+        $sessionToken = Yii::$app->session['access_token'];
+        $documentModel = new YiiDocumentModel();
+        $search["documentType"] = $this->documentFileType;
+        $search["concernedItem"] = $infrastructureUri;
+        //SILEX:info
+        // Order is "desc" by default
+        // $search["order"] = "desc"; can be used but it's not required
+        // in this case
+        //\SILEX:info
+        $wsResult = $documentModel->find($sessionToken, $search);
+        
+        // 3. Get the lastest acquisition session template saved
+        //    for the required document type
+        if ($wsResult != null && isset($wsResult[0])) {
+            $acquistionDocMetadata = $wsResult[0];
+            $existingFilePath = $documentModel->getDocument($sessionToken, $acquistionDocMetadata->uri, $acquistionDocMetadata->format);
+            //SILEX:conception
+            // Find a way to send a generic response when the user is disconnected
+            //\SILEX:conception
+            if ($existingFilePath == WSConstants::TOKEN) {
+                return $this->redirect(Yii::$app->urlManager->createUrl("site/login"));
+            }
+            // 4. Return the physical path of the file
+            $realPath = str_replace(\config::path()['documentsUrl'], Yii::getAlias('@webroot/documents/') , $existingFilePath );
+            return $realPath;
+        } 
+        
+        // 4. Return an warning if no file present
+        $this->error = $this->render(
+            SiteMessages::SITE_WARNING_PAGE_ROUTE, [
+                   SiteMessages::SITE_PAGE_NAME =>  SiteMessages::INTERNAL_ERROR,
+                   SiteMessages::SITE_PAGE_MESSAGE => SiteMessages::CANT_FETCH_FILE_AQUI_SESS
+            ]
+        );
+    }
+    
+    /**
+     * Put data retreived from the ws into the choosen worksheet
+     * @param type $existingFilePath the server physical path of the retrieved file 
+     * @return string
```

Comment:
Please explain the return because you return a filepath and often nothing "return;"

Reply:
Done